src/main/scala/beam/sim/BeamRunner.scala

```diff
@@ -0,0 +1,37 @@
+package beam.sim
+
+import scala.collection.JavaConverters._
+
+import com.google.inject
+import com.google.inject.Module
+import org.matsim.core.config.{Config => MatsimConfig}
+
+class BeamRunner(
+  val matsimConfig: MatsimConfig,
+  val outputDirectory: String,
+  val beamServices: BeamServices,
+  val injector: inject.Injector,
+  runMethod: () => Unit,
+  postRunner: BeamRunner => Any = _ => ()
+) {
+
+  def withPostRunner(func: BeamRunner => Any): BeamRunner = new BeamRunner(
+    matsimConfig = matsimConfig,
+    outputDirectory = outputDirectory,
+    beamServices = beamServices,
+    injector = injector,
+    runMethod = runMethod,
+    postRunner = func
+  )
+
+  def run(): BeamRunner = {
+    runMethod()
+    postRunner(this)
```

Comment:
This seems very cyclic. Is there a post run hook or could we make the runMethod go off of a Future?